src/uu/echo/src/echo.rs

```diff
@@ -23,63 +23,118 @@ mod options {
     pub const DISABLE_BACKSLASH_ESCAPE: &str = "disable_backslash_escape";
 }
 
-fn is_echo_flag(arg: &OsString) -> bool {
-    matches!(arg.to_str(), Some("-e" | "-E" | "-n"))
+/// Holds the parsed flags for echo command:
+/// -n (disable newline)
+/// -e/-E (escape handling),
+/// '-' (stop processing flags).
+struct EchoFlags {
+    /// -n flag: if true, don't output trailing newline
+    /// Default: false
+    pub disable_newline: bool,
+
+    /// -e enables escape interpretation, -E disables it
+    /// Default: false (escape interpretation disabled)
+    pub escape: bool,
+
+    /// Single hyphen '-' argument (stops flag processing)
+    /// Default: false
+    pub is_single_hyphen: bool,
 }
 
-// A workaround because clap interprets the first '--' as a marker that a value
-// follows. In order to use '--' as a value, we have to inject an additional '--'
-fn handle_double_hyphens(args: impl uucore::Args) -> impl uucore::Args {
-    let mut result = Vec::new();
-    let mut is_first_argument = true;
-    let mut args_iter = args.into_iter();
+/// Checks if an argument is a valid echo flag
+/// Returns EchoFlags if valid, None otherwise
+fn is_echo_flag(arg: &OsString) -> Option<EchoFlags> {
+    let bytes = arg.as_encoded_bytes();
+    if bytes.first() == Some(&b'-') {
+        let mut flags = EchoFlags {
+            disable_newline: false,
+            escape: false,
+            is_single_hyphen: false,
+        };
+        // Single hyphen case: "-" (stops flag processing, no other effect)
+        if arg.len() == 1 {
+            flags.is_single_hyphen = true;
+            return Some(flags);
+        }
```

Comment:
```suggestion
    if bytes.first() == Some(&b'-') && arg != "-" {
        let mut flags = EchoFlags {
            disable_newline: false,
            escape: false,
        };
```
Would this work? I think the `is_single_hyphen` case is identical to the `None` case

Reply:
Also correct. Changed this part too. Thanks!